core/src/main/java/io/questdb/cairo/TableUtils.java

```diff
@@ -1044,84 +1031,29 @@ public static int lengthOf(@Nullable CharSequence columnValue) {
     }
 
     public static @NotNull MatViewDefinition loadMatViewDefinition(
-            FilesFacade ff,
-            MemoryCMR mem,
+            MetaFileReader reader,
             Path path,
             int rootLen,
             TableToken matViewToken
     ) {
-        path.trimTo(rootLen)
-                .concat(matViewToken.getDirName())
-                .concat(MAT_VIEW_FILE_NAME);
-        mem.smallFile(ff, path.$(), MemoryTag.MMAP_DEFAULT);
-        if (mem.size() < MV_HEADER_SIZE + 3 * Long.BYTES + 3 * Integer.BYTES + Character.BYTES) {
-            throw CairoException.critical(0)
-                    .put("cannot read materialized view definition, file is too small [path=")
-                    .put(path)
-                    .put(']');
-        }
-
-        long offset = MV_HEADER_SIZE;
-
-        final CharSequence baseTableName = mem.getStrA(offset);
-        if (baseTableName == null || baseTableName.length() == 0) {
-            throw CairoException.critical(0)
-                    .put("base table name for materialized view is empty [view=")
-                    .put(matViewToken.getTableName())
-                    .put(']');
-        }
-        offset += Vm.getStorageLength(baseTableName);
-        final String baseTableNameStr = Chars.toString(baseTableName);
-
-        final long fromMicros = mem.getLong(offset);
-        offset += Long.BYTES;
-
-        final long toMicros = mem.getLong(offset);
-        offset += Long.BYTES;
-
-        final long samplingInterval = mem.getLong(offset);
-        offset += Long.BYTES;
-
-        final char samplingIntervalUnit = mem.getChar(offset);
-        offset += Character.BYTES;
-
-        final CharSequence timeZone = mem.getStrA(offset);
-        offset += Vm.getStorageLength(timeZone);
-        final String timeZoneStr = Chars.toString(timeZone);
-
-        final CharSequence timeZoneOffset = mem.getStrA(offset);
-        final String timeZoneOffsetStr = Chars.toString(timeZoneOffset);
-
-        path.trimTo(rootLen)
-                .concat(matViewToken.getDirName())
-                .concat(MAT_VIEW_QUERY_FILE_NAME);
-        mem.smallFile(ff, path.$(), MemoryTag.MMAP_DEFAULT);
-        if (mem.size() < Integer.BYTES) {
+        path.trimTo(rootLen).concat(matViewToken.getDirName()).concat(MatViewDefinition.MAT_VIEW_DEFINITION_FILE_NAME);
+        reader.of(path.$());
+        MetaFileReader.BlockCursor cursor = reader.getCursor();
+        if (cursor.hasNext()) {
+            // for now, we only have one message in the file
```

Comment:
This code won't work if someone tries downgrading from a newer version which created a new block type. Maybe we should iterate through the available blocks, search for the one we support while ignoring others?

Reply:
I'm going to merge the PR: let's continue the discussion here and maybe change things in the OG branch.